modules/light-clients/09-localhost/light_client_module_test.go

```diff
@@ -0,0 +1,320 @@
+			},
+			true,
+		},
+		{
+			"success: packet acknowledgement verification",
+			func() {
+				suite.chain.GetSimApp().GetIBCKeeper().ChannelKeeper.SetPacketAcknowledgement(suite.chain.GetContext(), mock.PortID, ibctesting.FirstChannelID, 1, ibctesting.MockAcknowledgement)
+
+				merklePath := commitmenttypes.NewMerklePath(host.PacketAcknowledgementPath(mock.PortID, ibctesting.FirstChannelID, 1))
+				merklePath, err := commitmenttypes.ApplyPrefix(suite.chain.GetPrefix(), merklePath)
+				suite.Require().NoError(err)
+
+				path = merklePath
+				value = ibctesting.MockAcknowledgement
+			},
+			true,
+		},
+		{
+			"invalid type for key path",
+			func() {
+				path = mock.KeyPath{}
+			},
+			false,
+		},
+		{
+			"key path has too many elements",
+			func() {
+				path = commitmenttypes.NewMerklePath("ibc", "test", "key")
+			},
+			false,
+		},
+		{
+			"no value found at provided key path",
+			func() {
+				merklePath := commitmenttypes.NewMerklePath(host.PacketAcknowledgementPath(mock.PortID, ibctesting.FirstChannelID, 100))
+				merklePath, err := commitmenttypes.ApplyPrefix(suite.chain.GetPrefix(), merklePath)
+				suite.Require().NoError(err)
+
+				path = merklePath
+				value = ibctesting.MockAcknowledgement
+			},
+			false,
+		},
+		{
+			"invalid value, bytes are not equal",
+			func() {
+				channel := channeltypes.NewChannel(
+					channeltypes.OPEN,
+					channeltypes.UNORDERED,
+					channeltypes.NewCounterparty(mock.PortID, ibctesting.FirstChannelID),
+					[]string{exported.LocalhostConnectionID},
+					mock.Version,
+				)
+
+				suite.chain.GetSimApp().GetIBCKeeper().ChannelKeeper.SetChannel(suite.chain.GetContext(), mock.PortID, ibctesting.FirstChannelID, channel)
+
+				merklePath := commitmenttypes.NewMerklePath(host.ChannelPath(mock.PortID, ibctesting.FirstChannelID))
+				merklePath, err := commitmenttypes.ApplyPrefix(suite.chain.GetPrefix(), merklePath)
+				suite.Require().NoError(err)
+
+				path = merklePath
+
+				// modify the channel before marshalling to value bz
+				channel.State = channeltypes.CLOSED
+				value = suite.chain.Codec.MustMarshal(&channel)
+			},
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			tc.malleate()
+
+			lightClientModule, found := suite.chain.GetSimApp().IBCKeeper.ClientKeeper.GetRouter().GetRoute(exported.LocalhostClientID)
+			suite.Require().True(found)
+
+			err := lightClientModule.VerifyMembership(
+				suite.chain.GetContext(),
+				exported.LocalhostClientID,
+				clienttypes.ZeroHeight(),
+				0, 0, // use zero values for delay periods
+				localhost.SentinelProof,
+				path,
+				value,
+			)
+
+			if tc.expPass {
+				suite.Require().NoError(err)
+			} else {
+				suite.Require().Error(err)
+			}
+		})
+	}
+}
+
+func (suite *LocalhostTestSuite) TestVerifyNonMembership() {
+	var path exported.Path
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expPass  bool
```

Comment:
same here for `expError`

Reply:
ditto, address in https://github.com/cosmos/ibc-go/issues/5959